L1Trigger/TrackFindingTracklet/interface/Tracklet.h

```diff
@@ -0,0 +1,567 @@
+    }
+
+    void setFitPars(double rinvfit,
+                    double phi0fit,
+                    double d0fit,
+                    double tfit,
+                    double z0fit,
+                    double chisqrphifit,
+                    double chisqrzfit,
+                    double rinvfitexact,
+                    double phi0fitexact,
+                    double d0fitexact,
+                    double tfitexact,
+                    double z0fitexact,
+                    double chisqrphifitexact,
+                    double chisqrzfitexact,
+                    int irinvfit,
+                    int iphi0fit,
+                    int id0fit,
+                    int itfit,
+                    int iz0fit,
+                    int ichisqrphifit,
+                    int ichisqrzfit,
+                    int hitpattern,
+                    const std::vector<L1TStub*>& l1stubs = std::vector<L1TStub*>());
+
+    std::string trackfitstr();
+
+    Track makeTrack(std::vector<L1TStub*> l1stubs);
+
+    Track* getTrack() {
+      assert(fpgatrack_ != 0);
+      return fpgatrack_;
+    }
+
+    bool fit() const { return ichisqrphifit_.value() != -1; }
+
+    int layer() const;
+    int disk() const;
+    int disk2() const;
+
+    bool isBarrel() const { return barrel_; }
+    bool isOverlap() const { return overlap_; }
+    int isDisk() const { return disk_; }
+
+    bool foundTrack(L1SimTrack simtrk, double phioffset);
+
+    void setTrackletIndex(int index);
+
+    int trackletIndex() const { return trackletIndex_; }
+
+    void setTCIndex(int index) { TCIndex_ = index; }
+
+    int TCIndex() const { return TCIndex_; }
+
+    int TCID() const { return TCIndex_ * (1 << 7) + trackletIndex_; }
+
+    int getISeed() const;
+    int getITC() const;
+
+    unsigned int PSseed() { return ((layer() == 1) || (layer() == 2) || (disk() != 0)) ? 1 : 0; }
+
+    unsigned int seedIndex() const { return seedIndex_; }
+
+    unsigned int calcSeedIndex() const;
+
+  private:
+    unsigned int seedIndex_;
+
+    // three types of trackletss
+    bool barrel_;
+    bool disk_;
+    bool overlap_;
+    bool triplet_;
+
+    Trklet::Stub* innerFPGAStub_;
+    Trklet::Stub* middleFPGAStub_;
+    Trklet::Stub* outerFPGAStub_;
+
+    L1TStub* innerStub_;
+    L1TStub* middleStub_;
+    L1TStub* outerStub_;
+
+    int trackletIndex_;
+    int TCIndex_;
+
+    //Tracklet track parameters
+    TrackPars<FPGAWord> fpgapars_;
+
+    TrackPars<double> trackpars_;
+    TrackPars<double> trackparsapprox_;
+
+    int projlayer_[4];
+    int projdisk_[5];
+
+    //Track parameters from track fit
+    TrackPars<FPGAWord> fpgafitpars_;
+    FPGAWord ichisqrphifit_;
+    FPGAWord ichisqrzfit_;
+
+    TrackPars<double> fitpars_;
+    double chisqrphifit_;
+    double chisqrzfit_;
+
+    TrackPars<double> fitparsexact_;
+    double chisqrphifitexact_;
+    double chisqrzfitexact_;
+
+    int hitpattern_;
+
+    Track* fpgatrack_;
```

Comment:
if this is owned, please use an `std::unique_ptr`

Reply:
now unique_ptr

Reply:
This has been changed